homeassistant/components/matrix/__init__.py

```diff
@@ -509,16 +558,28 @@ async def _send_message(
             target_rooms=target_rooms, message_type="m.room.message", content=content
         )
 
-        if (
-            data is not None
-            and (image_paths := data.get(ATTR_IMAGES, []))
-            and len(target_rooms) > 0
-        ):
-            image_tasks = [
-                self.hass.async_create_task(self._send_image(image_path, target_rooms))
-                for image_path in image_paths
-            ]
-            await asyncio.wait(image_tasks)
+        if data is not None and len(target_rooms) > 0:
+            tasks = []
+            if image_paths := data.get(ATTR_IMAGES, []):
+                image_tasks = [
+                    self.hass.async_create_task(
+                        self._send_image(image_path, target_rooms)
+                    )
+                    for image_path in image_paths
+                ]
+                tasks.extend(image_tasks)
+            if image_urls := data.get(ATTR_IMAGE_URLS, []):
+                image_url_tasks = [
+                    self.hass.async_create_task(
+                        self._send_image_url(
+                            image_url, target_rooms, data[ATTR_VERIFY_SSL]
+                        )
+                    )
+                    for image_url in image_urls
+                ]
+                tasks.extend(image_url_tasks)
+            if tasks:
+                await asyncio.wait(tasks)
```

Comment:
As in the original: `asyncio.wait` on an empty iterable is a safe no-op
```suggestion
            await asyncio.wait(tasks)
```